include/El/core/NumericTypeConversion.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef EL_NUMERICTYPECONVERSION_HPP
+#define EL_NUMERICTYPECONVERSION_HPP
+
+/*!
+  @file
+
+  Utilities for converting between numeric types (float, double, int, half).
+  This is non-trivial when using the half-precision type.
+
+  Allows conversions from type F to type T using To<T>(F const &), for example
+  To<half>(0) returns a half value of zero.
+*/
+
+#ifdef HYDROGEN_HAVE_HALF
+#include <half.hpp>
+
+template <typename F>
+struct Caster
```

Comment:
There already exists an infrastructure for `Caster` in `include/El/core/Element/decl.hpp`. That could be moved here or this could be moved there -- doesn't matter -- but they should use the same template.